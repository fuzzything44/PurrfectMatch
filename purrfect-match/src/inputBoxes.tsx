import React from 'react';
import "./ContactPage.css"

interface inputProps {
    prompt: string;
}

export class Input extends React.Component<inputProps, {}> {
    
    public render() {
        return (
        <div className="row">
            {this.props.prompt} 
            <div>
                <input type="text" name="fname" className="textBox"/>
            </div>
        </div>
        );
    }
}

export class Message extends React.Component<inputProps, {}> {
    
    public render() {
        return (
        <div className="row">
            {this.props.prompt} 
            <div>
                <textarea name="message" rows={20} cols={40} className="textBox message" placeholder = "Type your message here...">
                </textarea>
            </div>
        </div>
        );
    }
}

export class Send extends React.Component<{},{}> {

    public render() {
        return (
            <button id="send"> Send</button>
        );
    }
}