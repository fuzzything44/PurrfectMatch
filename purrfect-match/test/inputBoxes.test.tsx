import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Input, Select, Message, Send } from 'inputBoxes';

configure({ adapter: new Adapter() });


describe("Input tests", () => {
    let wrapper: ShallowWrapper<any, any, any>;
    beforeEach(() => {
        wrapper = shallow(<Input prompt="mockPrompt" error="" onchange={jest.fn()} />);
    });

    test("Snapshot test", () => {
        expect(wrapper).toMatchSnapshot();
        wrapper.setProps({ error: "mockErr " });
        expect(wrapper).toMatchSnapshot();
    });

    test("Updates correctly", () => {
        wrapper.instance().handleChange({ target: { value: "mockNewVal" } });
        expect(wrapper.instance().props.onchange).toHaveBeenCalledWith("mockNewVal");
    });
});

describe("Select tests", () => {
    let wrapper: ShallowWrapper<any, any, any>;
    beforeEach(() => {
        wrapper = shallow(<Select prompt="mockPrompt" options={["op1", "op2"]} error="" onchange={jest.fn()} />);
    });

    test("Snapshot test", () => {
        expect(wrapper).toMatchSnapshot();
        wrapper.setProps({ error: "mockErr " });
        expect(wrapper).toMatchSnapshot();
    });

    test("Updates correctly", () => {
        wrapper.instance().handleChange({ target: { value: "op2" } });
        expect(wrapper.instance().props.onchange).toHaveBeenCalledWith("op2");
    });
});

describe("Message tests", () => {
    let wrapper: ShallowWrapper<any, any, any>;
    beforeEach(() => {
        wrapper = shallow(<Message prompt="mockPrompt" error="" onchange={jest.fn()} />);
    });

    test("Snapshot test", () => {
        expect(wrapper).toMatchSnapshot();
        wrapper.setProps({ error: "mockErr " });
        expect(wrapper).toMatchSnapshot();
    });

    test("Updates correctly", () => {
        wrapper.instance().handleChange({ target: { value: "mockNewVal" } });
        expect(wrapper.instance().props.onchange).toHaveBeenCalledWith("mockNewVal");
    });
});

describe("Send tests", () => {
    let wrapper: ShallowWrapper<any, any, any>;
    beforeEach(() => {
        wrapper = shallow(<Send onclick={jest.fn()} />);
    });

    test("Snapshot test", () => {
        expect(wrapper).toMatchSnapshot();
        wrapper.setProps({ error: "mockErr " });
        expect(wrapper).toMatchSnapshot();
    });

    test("Updates correctly", () => {
        wrapper.instance().handleSubmit();
        expect(wrapper.instance().props.onclick).toHaveBeenCalledWith();
    });
});