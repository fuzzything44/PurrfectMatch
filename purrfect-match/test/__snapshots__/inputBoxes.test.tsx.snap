// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input tests Snapshot test 1`] = `
<div
  className="inputElem"
>
  <div
    className="label"
  >
    mockPrompt
  </div>
  <div
    className="inputField"
  >
    <input
      className="textBox"
      name="fname"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div />
</div>
`;

exports[`Input tests Snapshot test 2`] = `
<div
  className="inputElem"
>
  <div
    className="label"
  >
    mockPrompt
  </div>
  <div
    className="inputField"
  >
    <input
      className="textBox"
      name="fname"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="inputError"
  >
    mockErr 
  </div>
</div>
`;

exports[`Message tests Snapshot test 1`] = `
<div
  className="inputElem"
>
  <div
    className="label"
  >
    mockPrompt
  </div>
  <div
    className="inputField"
  >
    <textarea
      className="textBox message"
      cols={60}
      onChange={[Function]}
      placeholder="Type your message here..."
      rows={20}
    />
  </div>
  <div />
</div>
`;

exports[`Message tests Snapshot test 2`] = `
<div
  className="inputElem"
>
  <div
    className="label"
  >
    mockPrompt
  </div>
  <div
    className="inputField"
  >
    <textarea
      className="textBox message"
      cols={60}
      onChange={[Function]}
      placeholder="Type your message here..."
      rows={20}
    />
  </div>
  <div
    className="inputError"
  >
    mockErr 
  </div>
</div>
`;

exports[`Select tests Snapshot test 1`] = `
<div
  className="inputElem"
>
  <div
    className="label"
  >
    mockPrompt
  </div>
  <select
    onChange={[Function]}
    style={
      Object {
        "height": "1.5em",
        "width": "150px",
      }
    }
  >
    <option
      disabled={true}
      selected={true}
    >
       Select an option 
    </option>
    <option>
      op1
    </option>
    <option>
      op2
    </option>
  </select>
  <div />
</div>
`;

exports[`Select tests Snapshot test 2`] = `
<div
  className="inputElem"
>
  <div
    className="label"
  >
    mockPrompt
  </div>
  <select
    onChange={[Function]}
    style={
      Object {
        "height": "1.5em",
        "width": "150px",
      }
    }
  >
    <option
      disabled={true}
      selected={true}
    >
       Select an option 
    </option>
    <option>
      op1
    </option>
    <option>
      op2
    </option>
  </select>
  <div
    className="inputError"
  >
    mockErr 
  </div>
</div>
`;

exports[`Send tests Snapshot test 1`] = `
<div
  className="inputElem"
  id="mockId"
>
  <button
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;

exports[`Send tests Snapshot test 2`] = `
<div
  className="inputElem"
  id="mockId"
>
  <button
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;
